include/blmc_drivers/devices/spi_bus.hpp

```diff
@@ -0,0 +1,270 @@
+     */
+    ~SpiBus();
+
+    /**
+     * Output and status
+     */
+
+    /**
+     * @brief Get the measurements from the main board.
+     * 
+     * @param udriver_id is the index of the spi port on the control board
+     * @param index 
+     * @return std::shared_ptr<const MotorInterface::ScalarTimeseries> 
+     */
+    virtual std::shared_ptr<const MotorInterface::ScalarTimeseries>
+    get_measurement(
+      const size_t udriver_id,
+      const MotorBoardInterface::MeasurementIndex& index) const ;
+
+    /**
+     * @copydoc MotorBoardInterface::get_status()
+     * Inherited from MotorBoardInterface
+     */
+    virtual std::shared_ptr<const MotorBoardInterface::StatusTimeseries>
+    get_status(const size_t udriver_id) const;
+
+    /**
+     * input logs
+     */
+
+    /**
+     * @copydoc MotorBoardInterface::get_control()
+     * Inherited from MotorBoardInterface
+     */
+    virtual std::shared_ptr<const MotorInterface::ScalarTimeseries> get_control(
+      const size_t udriver_id,
+      const MotorBoardInterface::ControlIndex& index) const;
+
+    /**
+     * @copydoc MotorBoardInterface::get_command()
+     * Inherited from MotorBoardInterface
+     */
+    virtual std::shared_ptr<const MotorBoardInterface::CommandTimeseries>
+    get_command(const size_t udriver_id) const;
+
+    /**
+     * @copydoc MotorBoardInterface::get_sent_control()
+     * Inherited from MotorBoardInterface
+     */
+    virtual std::shared_ptr<const MotorInterface::ScalarTimeseries>
+    get_sent_control(
+        const size_t udriver_id,
+        const MotorBoardInterface::ControlIndex& index) const;
+
+    /**
+     * @copydoc MotorBoardInterface::get_sent_command()
+     * Inherited from MotorBoardInterface
+     */
+    virtual std::shared_ptr<const MotorBoardInterface::CommandTimeseries>
+    get_sent_command(const size_t udriver_id) const;
+
+    /**
+     * Setters
+     */
+    
+    /**
+     * @copydoc MotorBoardInterface::set_control()
+     * Inherited from MotorBoardInterface
+     */
+    virtual void set_control(const size_t udriver_id, const double& control,
+                             const MotorBoardInterface::ControlIndex& index);
+    
+    /**
+     * @copydoc MotorBoardInterface::set_command()
+     * Inherited from MotorBoardInterface
+     */
+    virtual void set_command(const size_t udriver_id, 
+                             const MotorBoardCommand& command);
+
+    /**
+     * @copydoc MotorBoardInterface::send_if_input_changed()
+     * Inherited from MotorBoardInterface. This particualr instance does not
+     * actually check if it is is a new command or control as the full status of
+     * the robot is exchange at every tick.
+     */
+    virtual void send_if_input_changed();
+
+    /**
+     * @brief return s only once board and motors are ready.
+     */
+    bool is_ready();
+
+    /**
+     * @brief Wait until the robot is ready.
+     */
+    void wait_is_ready();
```

Comment:
in CanBusMotorBoard this is called wait_until_ready, probably makes sense to make this consistent